conda_gitenv/deploy.py

```diff
@@ -125,50 +161,78 @@ def deploy_repo(repo, target, desired_env_labels=None):
 
-            # Create a latest tag that points to the most recently tagged environment.
+            # Create a latest tag that points to the most recently tagged
+            # environment.
             if env_tags.get(branch.name):
                 latest_tag = max(env_tags[branch.name],
                                  key=lambda t: t.commit.committed_date)
                 all_labelled_tags['latest'] = latest_tag.name
 
             # Only deploy environments that match the given pattern.
             labelled_tags = {}
-            if desired_env_labels is None:
-                desired_env_labels = ['*']
+            if env_labels is None:
+                env_labels = ['*']
             for label, tag in all_labelled_tags.items():
-                if any([fnmatch.fnmatch('{}/{}'.format(branch.name, label),
-                                        env_label) for env_label in desired_env_labels]):
+                item = '{}/{}'.format(branch.name, label)
+                match = [fnmatch(item, env_label)
+                         for env_label in env_labels]
+                if any(match):
                     labelled_tags[label] = tag
 
             for tag in set(labelled_tags.values()):
-                deploy_tag(repo, tag, target)
+                deploy_tag(repo, tag, target,
+                           api_user=api_user, api_key=api_key)
+
+            # Lock down the package cache files which may contain
+            # API credentials.
+            mode = stat.S_IRUSR | stat.S_IWUSR
+            pkg_cache_urls = os.path.join(target, PKG_CACHE_NAME, 'urls.txt')
+            if os.path.isfile(pkg_cache_urls):
+                os.chmod(pkg_cache_urls, mode)
 
+            pkg_cache_urls = os.path.splitext(pkg_cache_urls)[0]
+            if os.path.isfile(pkg_cache_urls):
+                os.chmod(pkg_cache_urls, mode)
+
+            mode = stat.S_IRUSR | stat.S_IWUSR | stat.S_IXUSR
             for label, tag in labelled_tags.items():
                 with Locked(os.path.join(target, label)):
                     deployed_name = tag.split('-', 2)[2]
                     label_target = deployed_name
                     label_location = os.path.join(target, branch.name, label)
 
                     if os.path.exists(label_location):
-                        # Unix only:
                         if os.readlink(label_location) != label_target:
                             os.remove(label_location)
                    
                     if not os.path.exists(label_location):
-                        print('Linking {}/{} to {}'.format(branch.name, label, tag))
+                        msg = 'Linking {}/{} to {} ({})'
+                        print(msg.format(branch.name, label,
+                                         label_target, tag))
                         os.symlink(label_target, label_location)
 
+                    # Lock down the conda-meta directory, which may contain
+                    # API credentials.
```

Comment:
Do API credentiials definitely go into the conda-meta? I had a look but I couldn't see where it would put it

Reply:
For `conda-gitenv` yes it does, but not necessarily for `conda-rpms`...